BankWallet/BankWallet/Core/Managers/AppVersionManager.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+class AppVersionManager {
+    private let systemInfoManager: ISystemInfoManager
+    private let localStorage: ILocalStorage
+
+    init(systemInfoManager: ISystemInfoManager, localStorage: ILocalStorage) {
+        self.systemInfoManager = systemInfoManager
+        self.localStorage = localStorage
+    }
+}
+
+extension AppVersionManager: IAppVersionManager {
+
+    func didFinishLaunching() {
+        let latestVersion = AppVersion(version: systemInfoManager.appVersion, date: Date())
+        var appVersions = localStorage.appVersions
+        guard let lastVersion = appVersions.last else {
+            localStorage.appVersions = [latestVersion]
+            return
+        }
+
+        if lastVersion.version != latestVersion.version {
```

Comment:
How AppVersion objects are compared here? It does not conform to Comparable protocol.
Date is always created as current date for "latestVersion" and it seems to be always not equal to latest saved version.